### DISCUSSION TITLE
Get array of ids instead of objects for relation

### DISCUSSION THREAD

**Discussion by Noitidart** [prisma/prisma#15680] [Q&A] [answered]
Is it possible to just get back array of id's? Rather then objects of a relation?

**Comment by austincrim** [comment, marked as answer]
Hey @Noitidart,

You can use the [`select`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#select) or [`include`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) options to specify you only want to retrieve the `id` field of the relation.

**Reply by Noitidart** [reply]
Thank you for fast reply @austincrim - however the best I could do with select/include can give us an array of objects that have the id. Is it possible to just get array of the id's?

**Reply by austincrim** [reply]
No, Prisma queries will always return objects.

**Reply by Noitidart** [reply]
Oh got it, thanks!

This is unrelated but while I got you, is there any difference between select/include?

**Reply by austincrim** [reply]
We explain some the different uses in [this section of our docs](https://www.prisma.io/docs/concepts/components/prisma-client/select-fields):

> By default, when a query returns records (as opposed to a count), the result includes the default selection set:
> - All scalar fields defined in the Prisma schema (including enums)
> - None of the relations
>
> To customize the result:
> - Use [select](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#select) to return specific fields - [you can also use a nested select to include relation fields](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#select-specific-relation-fields)
> - Use [include](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) to explicitly [include relations](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-reads)

**Reply by Noitidart** [reply]
Thanks very much sir! I still don't get the differences but I'll keep using it and hopefully it will make sense as I use it lol.

**Comment by jelmd** [comment]
If a query specifies that it should select a scalar field, only - it should be smart enough to return a single array of scalar values instead of a objects having just a single key,value pair.

**Comment by jelmd** [comment]
> Hey @Noitidart,
> 
> You can use the [`select`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#select) or [`include`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) options to specify you only want to retrieve the `id` field of the relation.

I doubt that this works. E.g. how would you write this not using an unsafe query?
```prisma
model Member {
    cid   Int       @mydb.Int // the ID of the club
    uid   Int       @mydb.Int // the ID of the user
    role  Role      // default role within the club
    inAt  DateTime  @default(now()) // when the user joined the team
    outAt DateTime? // when the user stopped being part of the team

    user User @relation(fields: [uid], references: [id], onDelete: Restrict)
    club Club @relation(fields: [cid], references: [id], onDelete: Restrict)

    @@id([cid, uid]) // primary key composed of both
    @@map(name: "member")
}
```
```ts
type numArray = {
    val: number[]
}

export async function findClubs(clubs: number[] | null, select: SELECT,
    forUID?: number, til?: Date)
{
    var count = 0;
    if (forUID && forUID > 0) {
        // get clubs the user is a member of, only.
        var q = 'SELECT JSON_ARRAYAGG(cid) AS val FROM member WHERE uid=' + forUID
            + ' AND (outAt IS NULL || outAt > "' + getDateTime() + '")';
        if (clubs && clubs.length > 0)
            q += ' AND cid IN(' + clubs.join(',') + ')';
        var res: numArray = (await prisma.$queryRawUnsafe<numArray[]>(q))[0];
        if (!res.val || res.val.length == 0)
            return null;
        clubs = res.val;
    }
    var clause: Record<string,any> = {};
    if (select != SELECT.ALL) {
        clause['active'] = select === SELECT.ENABLED;
        count++;
    }
    if (clubs && clubs.length > 0) {
        clause['id'] = { in: clubs };
        count++;
    }
    return count ? prisma.club.findMany({where: clause}) : prisma.club.findMany();
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
